### DISCUSSION TITLE
Allow only Yubi key for registration and authenication in UI

### DISCUSSION THREAD

**Discussion by Kyawsam** [MasterKale/SimpleWebAuthn#537] [Troubleshooting]
Is there any options to allow only USB key(Yubi Key) for both registration and authenication in the popup UI?

**Comment by MasterKale** [comment]
Hello @Kyawsam, check out the new **hints** that in the WebAuthn L3 draft:

https://w3c.github.io/webauthn/#dom-publickeycredentialcreationoptions-hints

For your use case you can try specifying `authenticatorAttachment: "cross-platform"` and `hints: ["security-key"]` in registration options and, in browsers that support hints, you'll see it start with a prompt to use a security key before anything else (though hints won't prevent a user from selecting other available options.)

Unfortunately hints aren't widely supported yet. If you want to test them out in various browsers then you can use https://webauthn.io and play around with hints in Advanced Settings:

![Screenshot 2024-04-13 at 9 59 25 AM](https://github.com/MasterKale/SimpleWebAuthn/assets/5166470/c04aac8e-a857-4dde-b455-fd0440d74a5d)

Aside from that there's not much else you can do to force the browser to only show security keys as an option...
